scala/src/main/scala/org/apache/spark/ml/recommendation/replay/ReplayALS.scala

```diff
@@ -0,0 +1,1824 @@
+/*
```

Comment:
we need to assure coping and rewriting the code is in accordance with the license,

Reply:
we do not have any NOTICE file in our repo, so the line 3 below becomes wrong.

have a look at section 4, which contains info about the NOTICE file:
https://www.apache.org/licenses/LICENSE-2.0

spark notice file is here: 
https://github.com/apache/spark/blob/master/NOTICE

could you bring your work up to the standard?